### PR TITLE
lint: add eslint-prettier

### PR DESCRIPTION
**Pull request by himself65**

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "trailingComma": "all",
+  "printWidth": 120,
+  "semi": false,
+  "singleQuote": true,
+  "jsxBracketSameLine": true
```
Comment (himself65) [inline .prettierrc:6]: add jsx check because for maybe the later use in the playground

Comment (Jack-Works) [inline .prettierrc:6]: We're not going to use JSX in this project

File: package.json [hunk 2]
```diff
@@ -32,6 +33,11 @@
     "typescript": "^4.3.2"
   },
   "dependencies": {
+    "eslint": "^7.29.0",
+    "eslint-config-prettier": "^8.3.0",
+    "eslint-plugin-import": "^2.23.4",
+    "eslint-plugin-prettier": "^3.4.0",
+    "eslint-plugin-unused-imports": "^1.0.0",
```
Comment (Jack-Works) [inline package.json:39]: And I guess this one is useless too.

File: package.json [hunk 3]
```diff
@@ -32,6 +33,11 @@
     "typescript": "^4.3.2"
   },
   "dependencies": {
+    "eslint": "^7.29.0",
+    "eslint-config-prettier": "^8.3.0",
+    "eslint-plugin-import": "^2.23.4",
```
Comment (Jack-Works) [inline package.json]: And I guess this one is useless too.

File: .prettierignore [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+.husky/
+package.json
+package-lock.json
+pnpm-lock.yaml
+
+*generated*
+storybook-static
+
+packages/storybook-shared/*.d.ts
+packages/storybook-shared/*.js
```
Comment (Jack-Works) [inline .prettierignore]: Seems like we don't need this part.

File: .prettierignore [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+
+# testing
+/reports
+/junit.xml
+
+# Build out
+dist/*
+/build
+/storybook-static
+/Maskbook.*.zip
```
Comment (Jack-Works) [inline .prettierignore]: Seems like we don't need this part.

File: .prettierignore [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+
+# E2E tests
+/.env/e2e-test
+/.env/e2e-development
+/.env/e2e-production
+/.pptr-alice
+/.pptr-bob
+/screenshots
+/.env
+/.pptr-*
```
Comment (Jack-Works) [inline .prettierignore]: Seems like we don't need this part.

File: .prettierignore [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+
+# Temp profiles
+.firefox
+.chrome
```
Comment (Jack-Works) [inline .prettierignore]: Seems like we don't need this part.

File: .prettierignore [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+# Rush temporary files
+common/deploy/
+common/temp/
+common/autoinstallers/*/.npmrc
+**/.rush/temp/
+
+## storybook-shared
+packages/storybook-shared/*.js
+packages/storybook-shared/*.map
+packages/storybook-shared/*.d.ts
```
Comment (Jack-Works) [inline .prettierignore]: Seems like we don't need this part.

**Review by Jack-Works — CHANGES_REQUESTED** [review]

File: .eslintrc.json [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+        "no-fallthrough": "error",
+        "no-new-wrappers": "error",
+        "no-plusplus": "error",
+        "no-restricted-globals": ["error", "event", "name", "length", "closed"],
+        "no-return-await": "error",
+        "no-sparse-arrays": "error",
+        "no-template-curly-in-string": "error",
+        "prefer-const": "warn",
+        "use-isnan": "error",
+        "unused-imports/no-unused-imports-ts": "error"
```
Comment (Jack-Works) [inline .eslintrc.json]: No need for this rule, it is somehow covered by the `importNotUsedAsValues` in `tsconfig`

**Review by Jack-Works — APPROVED** [review]

**Pull request merged**